app/dashboard/notifications.py

```diff
@@ -406,117 +407,92 @@ def build_github_notification(bounty, event_name, profile_pairs=None):
+        interested_context = []
+        started_context = []
+
         from_now = naturaltime(bounty.expires_date)
-        started_work = bounty.interested.filter(pending=False).all()
-        # pending_approval = bounty.interested.filter(pending=True).all()
         bounty_owner_clear = f"@{bounty.bounty_owner_github_username}" if bounty.bounty_owner_github_username else ""
-        approval_required = bounty.permission_type == 'approval'
-
-        if started_work.exists():
-            msg = f"{status_header}__Work has been started__.\n\n"
-        else:
-            msg = f"{status_header}__Workers have applied to start work__.\n\n"
-
-        msg += f"\nThese users each claimed they can complete the work by {from_now}. " \
-               "Please review their questions below:\n\n"
-
-        for i, interest in enumerate(interested, start=1):
-
-            profile_link = f"[{interest.profile.handle}]({interest.profile.url})"
-            action = "started work"
-            if interest.pending:
-                action = 'applied to start work'
-                action += f" _(Funders only: [approve worker]({bounty.approve_worker_url(interest.profile.handle)})"
-                action += f" | [reject worker]({bounty.reject_worker_url(interest.profile.handle)}))_"
-            if not interest.pending and approval_required:
-                action = 'been approved to start work'
-
-            show_dibs = interested.count() > 1 and bounty.project_type == 'traditional'
-            dibs = f" ({get_ordinal_repr(i)} dibs)" if show_dibs else ""
-
-            msg += f"\n{i}. {profile_link} has {action}{dibs}. "
-
-            issue_message = interest.issue_message.strip()
-            if issue_message:
-                msg += f"\t\n * Q: " \
-                       f"{issue_message}"
-            msg += "\n\n"
+        pending_interest = bounty.interested.filter(pending=True).select_related('profile').all()
+        if bounty.permission_type == 'approval':
+            for interest in pending_interest:
+               interested_context.append(
+                       SimpleNamespace(
+                           profile = interest.profile,
+                           approve_link = bounty.approve_worker_url(interest.profile.handle),
+                           reject_link = bounty.reject_worker_url(interest.profile.handle),
+                           issue_comment = interest.issue_message
+                        )
+                )
+            context = {
+                'bounty': bounty,
+                'interested': interested_context
+            }
+            
+            msg['pending'] = render_to_string(template_path + 'express_interest.md', context)
+        started_work = bounty.interested.filter(pending=False).select_related('profile').all()
+        if started_work.count():
+            for interest in started_work:
+                started_context.append(
+                        SimpleNamespace(
```

Comment:
Any reason for using `SimpleNamespace` for this over passing a django queryset in the template context?

Reply:
@mbeacom I was thinking django templates do not support arguments in their function calls, so we may not be able to pass the query set and call bounty.approve/reject_worker_url(interest.profile.handle) in the template. We could probably add a templatetag for that though

Reply:
It's supported.  Check out:  https://docs.djangoproject.com/en/2.1/ref/templates/language/ and https://docs.djangoproject.com/en/2.1/ref/templates/

Reply:
@mbeacom Hmmm on the first page you linked it says:
Because Django intentionally limits the amount of logic processing available in the template language, it is not possible to pass arguments to method calls accessed from within templates. Data should be calculated in views, then passed to templates for display.